test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java

```diff
@@ -409,6 +419,100 @@ public void narrativeDecoration() throws Exception {
+                SourceData.getBuilder(new UObject(new HashMap<>()), "objname2", "creator")
+                        .build(),
+                Instant.ofEpochMilli(10000),
+                null,
+                new GUID("WS:2/1/1"),
+                ImmutableMap.of(new GUID("WS:2/1/1"), new ParsedObject(
+                        "{\"whee\": \"imaprettypony1\"}",
+                        ImmutableMap.of("whee", Arrays.asList("imaprettypony1")))),
+                true);
+
+        final ObjectData expected1 = new ObjectData()
+                .withData(new UObject(ImmutableMap.of("whee", "imaprettypony1")))
+                .withGuid("WS:1/1/1")
+                .withKeyProps(ImmutableMap.of("whee", "imaprettypony1"))
+                .withCreator("creator")
+                .withType("Deco")
+                .withTypeVer(1L)
+                .withObjectName("objname1")
+                .withTimestamp(10000L);
+
+        final ObjectData expected2 = new ObjectData()
+                .withData(new UObject(ImmutableMap.of("whee", "imaprettypony1")))
+                .withGuid("WS:2/1/1")
+                .withKeyProps(ImmutableMap.of("whee", "imaprettypony1"))
+                .withCreator("creator")
+                .withType("Deco")
+                .withTypeVer(1L)
+                .withObjectName("objname2")
+                .withTimestamp(10000L);
+
+        SearchObjectsOutput noAuthsearchObjsOut;
+        SearchTypesOutput noAuthsearchTypesOut;
+        GetObjectsOutput noAuthGetObjsOut;
+
+        try {
+            // use the search client without authentication to get results
+            noAuthsearchObjsOut =  noAuthSearchCli.searchObjects(new SearchObjectsInput()
+                    .withAccessFilter(new AccessFilter())
+                    .withMatchFilter(new MatchFilter()));
+            noAuthsearchTypesOut = noAuthSearchCli.searchTypes(new SearchTypesInput()
+                    .withMatchFilter(new MatchFilter())
+                    .withAccessFilter(new AccessFilter()));
+            noAuthGetObjsOut = noAuthSearchCli.getObjects(new GetObjectsInput()
+                    .withGuids(Arrays.asList("WS:1/1/1", "WS:2/1/1")));
+
+        } catch (ServerException e) {
+            System.out.println("Exception server side trace:\n" + e.getData());
+            throw e;
+        }
+
+        assertThat("incorrect search objects count", noAuthsearchObjsOut.getObjects().size(), is(1));
+        TestCommon.compare(noAuthsearchObjsOut.getObjects().get(0), expected2);
+
+        Map<String, Long> typeToCount = noAuthsearchTypesOut.getTypeToCount();
+        assertThat("incorrect search types map size", typeToCount.size(), is(1));
+        assertThat("Incorrect search types type name", typeToCount.keySet(), is(set("Deco")));
+        assertThat("Incorrect search types type count", typeToCount.get("Deco"), is(Long.valueOf(1)));
+
+        assertThat("incorrect get objects count", noAuthGetObjsOut.getObjects().size(), is(0));
```

Comment:
I was expecting to see getObjects().size() = 1 because the public object with guid ws:2/1/1 should get returned. Am I missing something here?